### PR TITLE
Docker dev QoL updates

### LINKED ISSUE #1435: improve docker development experience
this issue covers a set of related pain points with the local development experience using docker:
- [x] stale `/opt/tmp/pids/server.pid` causes rails to think server is already running
- [x] missing gems on app container restart causes bundler error
- [x] solr core state is not persisted across restarts of compose stack
- [x] https://github.com/sul-dlss/dlme/issues/1437

### PR DESCRIPTION
**Pull request by thatbudakguy**
- Specify solr and postgres versions for local dev
- Reformat docker-compose file
- Persist solr data using named volume
- Use entrypoint script to run rails server for app
- Move args/env to beginning of docker builds
- Use envvar to tell bundler what dependencies to omit
- Use ENTRYPOINT command for worker build
- Add more docker info to README

## Why was this change made?
fixes #1435 

## Was the documentation (README, API, wiki, ...) updated?
yup

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,61 +25,66 @@ services:
...
       - ./solr/config:/solr-setup/conf
     entrypoint:
       - docker-entrypoint.sh
       - solr-precreate
       - dlme
       - /solr-setup/conf
+
   redis:
-    image: 'redis:4.0-alpine'
+    image: "redis:4.0-alpine"
```
Comment (cbeer) [inline docker-compose.yml:81]: AWS is currently using 5.x 🤷‍♂️ 
```suggestion
    image: "redis:5.0-alpine"
```

File: docker/dev-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -e
+
+# prevent bundle errors on missing gems
+bundle check || bundle install --jobs 20 --retry 5
```
Comment (cbeer) [inline docker/dev-entrypoint.sh:6]: I wonder if `./bin/setup` would be a useful addition here? It does the bundle check business too 🤷‍♂️

Comment (cbeer) [reply docker/dev-entrypoint.sh:6]: Ah, must have been a Rails 5.x thing. I think we should remove it in favor of setup?

File: README.md [hunk 3]
```diff
@@ -75,20 +75,39 @@ $ bundle exec rake spotlight:initialize
...
+```sh
+$ docker compose down
+```
+This process will gracefully stop and remove all running containers. Data is persisted in the form of volumes, and won't be removed. To start the stack back up again:
+```sh
+$ docker compose up     # add -d to run in the background and silence logs
+```
+You can stop and start individual containers by referring to them by name:
+```sh
+$ docker compose stop dlme-app-1    # use `docker ps` to see the container names
```
Comment (corylown) [inline README.md:98]: `docker compose stop` seems to want the name `app` and not `dlme-app-1`

File: README.md [hunk 4]
```diff
@@ -75,20 +75,39 @@ $ bundle exec rake spotlight:initialize
...
-$ docker compose run app bundle exec rake db:setup
-$ docker compose run app bundle exec rake spotlight:initialize
-[THEN]
-$ docker compose up -d
+#### Initial setup
+To get the development environment set up for the first time, you'll need to build the container images and prepare the database. You only need to do this once.
+```sh
+$ docker compose up --build
+$ docker compose run --rm app rake db:setup
+$ docker compose run --rm app rake spotlight:initialize
```
[the remark this answers is not in the document]
Comment (thatbudakguy) [reply README.md]: `rake db:setup` should no longer be necessary; documentation updated

**Review by marlo-longley — APPROVED** [review]
@thatbudakguy I was able to follow these instructions and get up and running. Thanks for all your work on this!

**Review by corylown — APPROVED** [review]
It's working for me, and the readme updates are thorough and very helpful.

**Review by camillevilla — APPROVED** [review]
Working for me! Persisting the solr data is 💯

**Pull request merged**